### PR TITLE
Add user_data_replace_on_change

### PR DESCRIPTION
**Pull request by bwmetcalf**
## what
* Add `user_data_replace_on_change` parameter to allow rebuilding of instances when user data changes.

## why
* Setting `user_data_replace_on_change` to `true` will restore old behavior.

## references
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#user_data_replace_on_change

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -117,6 +117,7 @@ resource "aws_instance" "default" {
   disable_api_termination              = var.disable_api_termination
   user_data                            = var.user_data
   user_data_base64                     = var.user_data_base64
+  user_data_replace_on_change          = var.user_data_replace_on_change
```
[the remark this answers is not in the document]
Comment (woz5999) [reply main.tf:120]: It looks like you may need to add a comment to skip this check since it seems unrelated to your changes.

**Review by woz5999 — APPROVED** [review]
lgtm

**Review by Nuru — CHANGES_REQUESTED** [review]
Among other things, let's not require AWS provider v4 yet.

**Comment by bwmetcalf** [comment]
> Among other things, let's not require AWS provider v4 yet.

OK.  4.7.0 of the provider is the earliest version that supports  `user_data_replace_on_change`, so I'll leave this open for now.

**Comment by Gowiem** [comment]
@bwmetcalf want to work through the conflicts and rebuild README (via `make init && make readme`) and we'll work on this one? I saw you comment `/open` and this looks like a legit change, so I saw we push it forward. Make those changes and ping me when you're ready!

**Pull request closed without merging**